### ISSUE TITLE
Panel objects should mirror bokehjs convention

### ISSUE THREAD

**Issue by hyamanieu** [holoviz/panel#2834]
#### ALL software version info
panel 0.12.4
bokeh 2.4.1

#### Description of expected behavior and the observed behavior

I am expecting that the `name` parameter of a Panel object corresponds to the name attribute of the equivalent bokeh model, but it is renamed in

https://github.com/holoviz/panel/blob/9966691fe18f0a492566f540e2661792e3e75fa1/panel/widgets/base.py#L33

It becomes impossible within `jscallback`s to use `document.getElementsbyName`.

same thing with `tags`, which isn't set at all.

IMHO, this is a bug more than a lack of feature.

#### Complete, minimal, self-contained example code that reproduces the issue

```
import panel as pn
pn.extension()

ti = pn.widgets.TextInput(name='foo')

ti.jscallback(value='console.log(cb_obj.name)')

ti

```

open the console log and you'll see that `null` is returned.

#### Stack traceback and/or browser JavaScript console output

```javascript
null
```

#### proposed solution

I would let the parameters `name` be its former self, add `tags` and `title` ; then let `title`  & `name` parameters sync with each other unless they are explicitly set upon instantiating the widget

However my knowledge of the library is not deep enough to understand what would be the effects downstream.

**Comment by philippjfr** [comment]
I strongly agree with this and it **may** be considered for Panel 1.0. However I have a strong inclination this ship has sailed as it would affect far too much user code at this point.

**Comment by hyamanieu** [comment]
Changes in the user guide & documentation galery:
- I have added the `tags` attribute to all widget docs
- I have replaced `name ` wherever necessary by `title` or `label`, or removed it entirely.
- I have removed entirely `name`/`title`/`label` below the "Display" chapter if setting it had no influence on what is displayed
- I have mentioned `name` under the "Core" chapter only if it is available at the DOM element level, which is not the case for many widgets, cf. https://github.com/bokeh/bokeh/issues/11782 (I still have plenty to learn)

Still issues with testing.
